### PR TITLE
[AMORO-3931] Exclude curator and zookeeper deps

### LINKED ISSUE #3931: [Subtask]: Exclude curator and zookeeper deps
### Search before asking

- [x] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

#3930 

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by turboFei**
## Why are the changes needed?

Close #3931.

Given that, amoro already shade the zookeeper and curator for internal using, https://github.com/apache/amoro-shade/tree/main/amoro-shade-zookeeper-3.

We can exclude them to reduce the conflicts.
## Brief change log

- Exclude all curator and zookeeper deps

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by turboFei** [comment]
cc @zhoujinsong @xxubai

**Comment by xxubai** [comment]
Can you talk about why we need to exclude zk&curator dependencies?

**Comment by turboFei** [comment]
Have updated the PR desc.

Given that, amoro already shade the zookeeper and curator for internal using, https://github.com/apache/amoro-shade/tree/main/amoro-shade-zookeeper-3.

We can exclude them to reduce the conflicts.

@xxubai

**Comment by turboFei** [comment]
they are all transitive dependencies, if we did need them, we shall define them in dependencyManagement.

**Comment by turboFei** [comment]
now the transitive dependencies are undertermined, it might cause 
- conflicts
- CVE

File: pom.xml [hunk 1]
```diff
@@ -672,6 +710,48 @@
                 <version>${kryo.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.apache.hive</groupId>
+                <artifactId>hive-exec</artifactId>
+                <version>${hive.version}</version>
+                <classifier>core</classifier>
+                <scope>test</scope>
```
Comment (xxubai) [inline pom.xml]: nit: move to test deps block

Comment (turboFei) [reply pom.xml]: updated

Comment (turboFei) [reply pom.xml]: I removed the scope in dependencyManagement, we may need it, will followup it in another pr

https://github.com/apache/amoro/issues/3933

Comment (xxubai) [reply pom.xml]: What I mean is that this is a test dependency, and as noted below, all dependencies with the test scope should be placed there.
```
<!-- test dependencies -->
...
```

Comment (xxubai) [reply pom.xml]: I see. it is not in the dependencyManagement.

Comment (turboFei) [reply pom.xml]: It should be fine to remove the test scope in the dependencyManagement. 

image

File: amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/pom.xml [hunk 2]
```diff
@@ -394,28 +394,6 @@
             <version>${hive.version}</version>
             <classifier>core</classifier>
             <scope>test</scope>
```
Comment (turboFei) [inline amoro-format-mixed/amoro-mixed-spark/v3.3/amoro-mixed-spark-3.3/pom.xml:396]: we reserved the scope in submodules

File: amoro-format-mixed/amoro-mixed-spark/v3.5/amoro-mixed-spark-3.5/pom.xml [hunk 3]
```diff
@@ -398,28 +398,6 @@
             <version>${hive.version}</version>
             <classifier>core</classifier>
             <scope>test</scope>
```
Comment (turboFei) [inline amoro-format-mixed/amoro-mixed-spark/v3.5/amoro-mixed-spark-3.5/pom.xml:400]: it has scope

File: amoro-format-mixed/amoro-mixed-hive/pom.xml [hunk 4]
```diff
@@ -53,20 +53,6 @@
             <version>${hive.version}</version>
             <classifier>core</classifier>
             <scope>test</scope>
```
Comment (turboFei) [inline amoro-format-mixed/amoro-mixed-hive/pom.xml:55]: also here

**Review by xxubai — APPROVED** [review]
